src/distributions/normal.jl

```diff
@@ -57,11 +59,7 @@ Computes a Normal distribution of eltype T from the Normal distribution N if T a
 That is T and U must both be Real or both be Complex.
 """
 function Normal{T}(N::Normal{U,V,W}) where {T,U,V<:AbstractVector,W<:CovarianceParameter}
-    T <: Real && U <: Real || T <: Complex && U <: Complex ?
-    Normal(convert(AbstractVector{T}, N.μ), convert(CovarianceParameter{T}, N.Σ)) :
-    error(
-        "The constructor type $(T) and the argument type $(U) must both be real or both be complex",
-    )
+    return Normal(convert(AbstractVector{T}, N.μ), convert(CovarianceParameter{T}, N.Σ))
```

Comment:
As in other places, why do the conversion here?

Reply:
I think it is easier to reason about what the return type is from certain functions if a uniform precision is enforced on the data. Can you think of a case where you really want to have a non-uniform precision?

Reply:
Maybe there could be an argument to why you want covariances and means to have different precisions because one is basically the square of the other?

But I guess my main question is why it is important to as the reader infer the return type from certain functions? I normally expect that as long as my functions are generic enough they should be type-stable and performant?

Reply:
I am sure we can have type stable and performant code with mixed precision data as well. What I mean is that enforcing a uniform type leads to a complexity reduction for me when developing. It also leads to needing less tests as we dont have to make sure everything works as expected for every combination of eltypes in mean and covariance. 

In any case I feel this issue is a bit orthogonal to the present PR. Would it make sense to discuss this in a separate issue?

Reply:
I think I was mainly surprised that this question is so relevant for the development process of the code, as I would have expected that you do not actually have to think about which types could be in there, and I was curious to where this pops up.

Sure, we can track this in a separate issue if you prefer.